### PR TITLE
HAWQ-25. Add resource queue new ddl statement implementation and refi…

### PR DESCRIPTION
**Pull request by jiny2**
…ne partial GUC variables

### CODE REVIEW

File: src/backend/commands/user.c [hunk 1]
```diff
@@ -554,31 +541,34 @@ CreateRole(CreateRoleStmt *stmt)
...
 	{
-		if (strcmp(resqueue, "none") == 0) {
+		if (strcmp(resqueue, "none") == 0)
+		{
 			unregisterConnectionInRMWithErrorReport(resourceid);
 			releaseResourceContextWithErrorReport(resourceid);
 			ereport(ERROR,
 					(errcode(ERRCODE_RESERVED_NAME),
-					 errmsg("resource queue name \"%s\" is reserved",
+					 errmsg("Resource queue name \"%s\" is reserved",
```
Comment (jianlirong) [inline src/backend/commands/user.c]: According to the PostgreSQL Error Message Style Guide, "Do not capitalize the first letter". I think here the old one is better than the new one.

Comment (jiny2) [inline src/backend/commands/user.c]: Accepted.

**Comment by zhangh43** [comment]
changes in cdbdatalocality.c is OK.

**Comment by linwen** [comment]
OK.

File: src/backend/commands/user.c (hunk 1 above)
Comment (cwelton) [inline src/backend/commands/user.c]: I agree with Lirong, the old message conforms to error message guidelines and is preferable to the replacement.

File: src/backend/utils/misc/guc.c [hunk 2]
```diff
@@ -8003,14 +8024,14 @@ static struct config_string ConfigureNamesString[] =
...
+    {
+		{"hawq_global_rm_type", PGC_POSTMASTER, RESOURCES_MGM,
+				gettext_noop("set resource management server type"),
+				NULL
 		},
-		&rm_grm_server_type,
+		&rm_global_rm_type,
 		"none", NULL, NULL
-	},
+    },
```
Comment (amyrazz44) [inline src/backend/utils/misc/guc.c]: changes in guc.c is ok.

**Comment by jianlirong** [comment]
Except these issues mentioned before by Caleb and myself, the logic of the modifications looks good to me.

**Comment by jiny2** [comment]
I have fixed the errormessage comments and renamed queue name column of pg_resqueue from "name" to "rsqname".

**Comment by jiny2** [comment]
This is a big code merge for adding recent code work from old hawq repository. This merge mainly includes

1) implement new CREATE/ALTER RESOURCE QUEUE attributes;
2) refine partial GUC variable names;
3) use new libyarn lib with kerberbos support.

**Comment by cwelton** [comment]
> 1) implement new CREATE/ALTER RESOURCE QUEUE attributes;

I went to look for what these changes were so that I could update the jira description accordingly and noticed that you haven't updated {{./doc/src/sgml/ref/create_resource_queue.sgml}}, which means that the command line help won't be accurate in psql.

**Pull request closed without merging**